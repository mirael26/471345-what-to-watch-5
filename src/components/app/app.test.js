import React from "react";
import renderer from "react-test-renderer";

import {Router} from "react-router-dom";
import browserHistory from "../../browser-history";

import App from "./app";

jest.mock(`../main/main`, () => `Main`);
jest.mock(`../sign-in/sign-in`, () => `SignIn`);
jest.mock(`../my-list/my-list`, () => `MyList`);
jest.mock(`../movie/movie`, () => `Movie`);
jest.mock(`../player/player`, () => `Player`);
jest.mock(`../add-review/add-review`, () => `AddReview`);
jest.mock(`../private-route/private-route`, () => `PrivateRoute`);
jest.mock(`../error-page/error-page`, () => `ErrorPage`);

it(`Should App render correctly`, () => {
  const tree = renderer
    .create(
        <Router history={browserHistory}>
          <App/>
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
