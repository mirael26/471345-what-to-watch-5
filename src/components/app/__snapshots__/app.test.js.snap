// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should App render correctly 1`] = `
<Main
  onPlayerButtonClick={[Function]}
  onUnauthorizedFavoriteClick={[Function]}
/>
`;
